resources/2023-06-resource-extensibility-product-spec.md

```diff
@@ -0,0 +1,133 @@
+
+### Scenario 3: Raj publishes Recipes for proprietary services
+Raj is a partner or a system integrator who helps enterprises integrate proprietary service for eg : Oracle database in their cloud native applications. Raj tries out Radius and wants to write Recipes to create an Oracle Database with a standard set of best practices and policies integrated. He works with a lot of customers who are wanting to use the Oracle Database recipe in their applications.
+
+## Key dependencies and risks
+<!-- What dependencies must we take in order to enable this scenario? -->
+
+<!-- What other risks are you aware of that need to be mitigated. If you have a mitigation in mind, summarize here. -->
+<!-- **Dependency Name** – summary of dependency.  Issues/concerns/risks with this dependency -->
+<!-- **Risk Name** – summary of risk.  Mitigation plan if known. If it is not yet known, no problem. --> 
+
+- Dependency: Bicep compiler merge support
+    - We need to ensure that the ongoing Bicep compiler merge work is complete so that custom resource types can be dynamically generated for the user defined resource types.
+    - Mitigation: Work closely with the Bicep team to ensure that the merge is completed on time and that the necessary features are available to enable custom resource types in Radius .
+
+- Risk: Adoption and community engagement
+    - There is a risk that users may not fully embrace the extensibility feature or contribute custom resource types to the community.
+    - Mitigation: Actively engage with users, provide clear documentation and resources, and foster a supportive and inclusive community to encourage adoption and contributions.
+
+- Risk: Compatibility and maintainability
+    - As more custom resource types are added to Radius, there is a risk of compatibility issues and increased maintenance overhead.
+    - Mitigation: Implement thorough testing and versioning strategies to ensure compatibility, and establish clear guidelines for maintaining custom resource types.
+
+- Risk: Security and trust
+    - Allowing users to contribute custom resource types introduces potential security risks and the need to establish trust in the community-contributed code.
+    - Mitigation: Implement strict security measures, such as code reviews and vulnerability scanning, and establish a transparent review process for community contributions.
+
+## Key assumptions to test and questions to answer
+<!-- If you are making assumptions that, if incorrect, would cause us to significantly alter our approach to this scenario, make them explicit here.  Also call out how / when you plan to validate key assumptions. -->
+<!-- What big questions must we answer in order to clarify our plan for this scenario.  When and how do you plan to answer those questions (prototype feature x, customer research, competitive research, etc) -->
+
+- Assumption: Users will find value in the extensibility feature and actively contribute to the community.
```

Comment:
one more assumption: users will use extensibility in lieu of existing extenders because they want to take advantage of things like strong typing and validation in Radius.